app/src/main/java/com/liato/bankdroid/DataRetrieverTask.java

```diff
@@ -68,37 +72,74 @@ public DataRetrieverTask(final MainActivity parent, final long bankId) {
 
     @Override
     protected void onPreExecute() {
-        this.dialog.setMessage(res.getText(R.string.updating_account_balance)
+        getDialog().setMessage(res.getText(R.string.updating_account_balance)
                 + "\n ");
-        this.dialog.setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
-        this.dialog.setCancelable(false);
-        this.dialog.show();
+        getDialog().setProgressStyle(ProgressDialog.STYLE_HORIZONTAL);
+        getDialog().setCancelable(false);
+        getDialog().show();
+    }
+
+    @NonNull
+    protected ProgressDialog getDialog() {
+        return this.dialog;
+    }
+
+    protected Bank getBankFromDb(long bankId, Context parent, boolean loadAccounts) {
+        return BankFactory.bankFromDb(bankId, parent, true);
+    }
+
+    protected List<Bank> getBanksFromDb(Context parent, boolean loadAccounts) {
+        return BankFactory.banksFromDb(parent, true);
+    }
+
+    protected void saveBank(Bank bank, Context context) {
+        DBAdapter.save(bank, parent);
+    }
+
+    protected void publishProgress(int i, @Nullable Bank bank) {
+        String text = "";
+        if (bank != null) {
+            text = bank.getName() + " (" + bank.getUsername() + ")";
+        }
+        publishProgress(Integer.toString(i), text);
+    }
+
+    protected boolean isContentProviderEnabled() {
+        final SharedPreferences prefs = PreferenceManager
+                .getDefaultSharedPreferences(parent);
+        return prefs.getBoolean("content_provider_enabled", false);
     }
 
     @Override
     protected Void doInBackground(final String... args) {
         errors = new ArrayList<String>();
-        ArrayList<Bank> banks;
+        List<Bank> banks;
         if (bankId != -1) {
             banks = new ArrayList<Bank>();
-            banks.add(BankFactory.bankFromDb(bankId, parent, true));
+            banks.add(getBankFromDb(bankId, parent, true));
         } else {
-            banks = BankFactory.banksFromDb(parent, true);
+            banks = getBanksFromDb(parent, true);
         }
         bankcount = banks.size();
-        this.dialog.setMax(bankcount);
+        getDialog().setMax(bankcount);
         int i = 0;
         for (final Bank bank : banks) {
-            publishProgress(new String[]{new Integer(i).toString(),
-                    bank.getName() + " (" + bank.getUsername() + ")"});
-            if (bank.isDisabled()) {
+            publishProgress(i, bank);
+
+            if (bank.isDisabled() && bankId == -1) {
```

Comment:
What do you think about moving out `bank.isDisabled && bankId == -1` to a separate method with a more descriptive name. If the method name is clear enough, we do not need the comments below.